crates/re_ui/src/list_item.rs

```diff
@@ -0,0 +1,196 @@
+    re_ui: &'a ReUi,
+    active: bool,
+    selected: bool,
+    icon_fn:
+        Option<Box<dyn FnOnce(&ReUi, &mut egui::Ui, egui::Rect, egui::style::WidgetVisuals) + 'a>>,
+    buttons_fn: Option<Box<dyn FnOnce(&ReUi, &mut egui::Ui) -> egui::Response + 'a>>,
+}
+
+impl<'a> ListItem<'a> {
+    /// Create a new [`ListItem`] with the given label.
+    pub fn new(re_ui: &'a ReUi, text: impl Into<egui::WidgetText>) -> Self {
+        Self {
+            text: text.into(),
+            re_ui,
+            active: true,
+            selected: false,
+            icon_fn: None,
+            buttons_fn: None,
+        }
+    }
+
+    /// Set the active state the item.
+    pub fn active(mut self, active: bool) -> Self {
+        self.active = active;
+        self
+    }
+
+    /// Set the selected state of the item.
+    pub fn selected(mut self, selected: bool) -> Self {
+        self.selected = selected;
+        self
+    }
+
+    /// Provide an [`Icon`] to be displayed on the left of the item.
+    pub fn with_icon(self, icon: &'a Icon) -> Self {
+        self.with_icon_fn(|re_ui, ui, rect, visuals| {
+            let image = re_ui.icon_image(icon);
+            let texture_id = image.texture_id(ui.ctx());
+            let tint = visuals.fg_stroke.color;
+
+            ui.painter().image(
+                texture_id,
+                rect,
+                egui::Rect::from_min_max(egui::pos2(0.0, 0.0), egui::pos2(1.0, 1.0)),
+                tint,
+            );
+        })
+    }
+
+    /// Provide a custom closure to draw an icon on the left of the item.
+    pub fn with_icon_fn(
+        mut self,
+        icon_fn: impl FnOnce(&ReUi, &mut egui::Ui, egui::Rect, egui::style::WidgetVisuals) + 'a,
+    ) -> Self {
+        self.icon_fn = Some(Box::new(icon_fn));
+        self
+    }
+
+    /// Provide a closure to display on-hover buttons on the right of the item.
+    pub fn with_buttons(
+        mut self,
+        buttons: impl FnOnce(&ReUi, &mut egui::Ui) -> egui::Response + 'a,
+    ) -> Self {
+        self.buttons_fn = Some(Box::new(buttons));
+        self
+    }
+
+    /// Draw the item.
+    pub fn show(self, ui: &mut Ui) -> Response {
+        ui.add(self)
+    }
+}
+
+impl egui::Widget for ListItem<'_> {
+    fn ui(self, ui: &mut Ui) -> Response {
+        let button_padding = ui.spacing().button_padding;
+        let icon_width_plus_padding = ReUi::small_icon_size().x + ReUi::text_to_icon_padding();
+        let icon_extra = if self.icon_fn.is_some() {
+            icon_width_plus_padding
+        } else {
+            0.0
+        };
+        let padding_extra = button_padding + button_padding;
+        let wrap_width = ui.available_width() - padding_extra.x - icon_extra;
+
+        let text =
+            self.text
+                .clone()
+                .into_galley(ui, Some(false), wrap_width, egui::TextStyle::Button);
+
+        let desired_size = (padding_extra + text.size() + egui::vec2(icon_extra, 0.0))
```

Comment:
nit: the icon is to the left of the text

```suggestion
        let desired_size = (padding_extra + egui::vec2(icon_extra, 0.0) + text.size())
```